kratos/sources/model_part_io.cpp

```diff
@@ -3637,6 +3639,29 @@ void  ModelPartIO::ReadSubModelPartConditionsBlock(ModelPart& rMainModelPart, Mo
     KRATOS_CATCH("")
 }
 
+void  ModelPartIO::ReadSubModelPartGeometriesBlock(ModelPart& rMainModelPart, ModelPart& rSubModelPart)
+{
+    KRATOS_TRY
+
+    SizeType geometry_id;
+    std::string word;
+    std::vector<SizeType> ordered_ids;
+
+    while (!mpStream->eof())
+    {
+        ReadWord(word); // Reading the node id or End
+        if (CheckEndBlock("SubModelPartGeometries", word))
+            break;
+
+        ExtractValue(word, geometry_id);
+                    ordered_ids.push_back(geometry_id);
+    }
+    std::sort(ordered_ids.begin(), ordered_ids.end());
```

Comment:
Why the sort?

This makes it inconsistent with the mdpa file

Or is this also done for elems and conds?

Reply:
Basically I took what we do for elements and conditions and I applied here as well.

Reply:
I've been checking the `AddGeometries` of the `ModelPart` and I don't see why the `std::sort` is needed. As far as I understand it shouldn't affect the performance of the key (id) search that we do in there as the underlying geometry container is an `std::unordered_map`.

Reply:
Seems that you didn't remove it but up to you

Reply:
Yes, I want to confirm the removal with the @KratosMultiphysics/technical-committee guys.